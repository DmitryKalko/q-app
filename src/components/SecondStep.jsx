import React, { useEffect } from "react";
import Box from "@mui/material/Box";
import InputLabel from "@mui/material/InputLabel";
import MenuItem from "@mui/material/MenuItem";
import FormControl from "@mui/material/FormControl";
import Select from "@mui/material/Select";
import Checkbox from "@mui/material/Checkbox";
import FormGroup from "@mui/material/FormGroup";
import FormControlLabel from "@mui/material/FormControlLabel";

import data from "../store/data";
import { observer } from "mobx-react-lite";

import useIsMobile from "../hooks/useIsMobile";

const SecondStep = observer(({ toNextStep, toPrevStep, lightTheme }) => {
  const isMobile = useIsMobile();

  const setGender = (e) => {
    data.setGender(e.target.value);
  };
  const setPolicyAgree = (e) => {
    data.setPolicyAgree(e.target.checked);
  };

  return (
    <div className="second-step">
      <Box sx={isMobile ? { width: 100 } : { width: 300 }}>
        <FormControl fullWidth>
          <InputLabel id="demo-simple-select-label">Gender</InputLabel>
          <Select
            labelId="demo-simple-select-label"
            id="demo-simple-select"
            value={data.store.gender}
            label="Gender"
            onChange={setGender}
            sx={lightTheme ? { color: "white" } : { color: "#6d6d6d" }}
          >
            <MenuItem value="Male">Male</MenuItem>
            <MenuItem value="Female">Female</MenuItem>
            <MenuItem value="Other">Other</MenuItem>
          </Select>
        </FormControl>
      </Box>
      <FormGroup>
        <FormControlLabel
          control={
            <Checkbox
              sx={{ color: "#50c2b7" }}
              onChange={setPolicyAgree}
              checked={data.store.policyAgree}
            />
          }
          label="I agree with the site policy"
          sx={lightTheme ? { color: "white" } : { color: "#6d6d6d" }}
        />
      </FormGroup>
      <div className="btns-block">
        <button
          className="btn"
          onClick={toPrevStep}
          style={lightTheme ? { color: "white" } : { color: "#6d6d6d" }}
        >
          BACK
        </button>
        <button
          className="btn"
          onClick={toNextStep}
          style={lightTheme ? { color: "white" } : { color: "#6d6d6d" }}
        >
          NEXT
        </button>
      </div>
    </div>
  );
});

export default SecondStep;
