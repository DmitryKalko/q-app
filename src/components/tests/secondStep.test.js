import { shallow } from "enzyme";
import React from "react";
import SecondStep from "../SecondStep";


const setUp = () => shallow(<SecondStep  />);

describe('SecondStep component', () => {
    let component;

    beforeEach(() => {
        component = setUp(); 
    })
    test('should render SecondStep component', () => {
        expect(component).toMatchSnapshot();
    });


    it('SecondStep inputs', () => {
        const testState = { male: 'man', agree: true };

        expect(component.find('#demo-simple-select').prop('value')).toEqual('');
        //expect(component.find('#checkbox-agree').prop('value')).toEqual('');
        component.find('#demo-simple-select').simulate('change', { target: { male: 'man' } });
        //component.find('#checkbox-agree').simulate('change', { target: { agree: true } });  //  не находит checkbox-agree т к shallow(<SecondStep  /> рендерит неглубоко - че делать?
        expect(testState.male).toEqual('man');
        expect(testState.agree).toEqual(true);
      });
})