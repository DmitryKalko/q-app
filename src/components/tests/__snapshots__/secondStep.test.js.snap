// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondStep component should render SecondStep component 1`] = `
<div
  className="second-step"
>
  <ForwardRef(Box)
    sx={
      Object {
        "width": 300,
      }
    }
  >
    <ForwardRef(FormControl)
      fullWidth={true}
    >
      <ForwardRef(InputLabel)
        id="demo-simple-select-label"
      >
        Gender
      </ForwardRef(InputLabel)>
      <ForwardRef(Select)
        id="demo-simple-select"
        label="Gender"
        labelId="demo-simple-select-label"
        onChange={[Function]}
        sx={
          Object {
            "color": "#6d6d6d",
          }
        }
        value=""
      >
        <ForwardRef(MenuItem)
          value="Male"
        >
          Male
        </ForwardRef(MenuItem)>
        <ForwardRef(MenuItem)
          value="Female"
        >
          Female
        </ForwardRef(MenuItem)>
        <ForwardRef(MenuItem)
          value="Other"
        >
          Other
        </ForwardRef(MenuItem)>
      </ForwardRef(Select)>
    </ForwardRef(FormControl)>
  </ForwardRef(Box)>
  <ForwardRef(FormGroup)>
    <ForwardRef(FormControlLabel)
      control={
        <ForwardRef(Checkbox)
          checked={false}
          id="checkbox-agree"
          onChange={[Function]}
          sx={
            Object {
              "color": "#50c2b7",
            }
          }
        />
      }
      label="I agree with the site policy"
      sx={
        Object {
          "color": "#6d6d6d",
        }
      }
    />
  </ForwardRef(FormGroup)>
  <div
    className="btns-block"
  >
    <button
      className="btn"
      style={
        Object {
          "color": "#6d6d6d",
        }
      }
    >
      BACK
    </button>
    <button
      className="btn"
      style={
        Object {
          "color": "#6d6d6d",
        }
      }
    >
      NEXT
    </button>
  </div>
</div>
`;
